network/src/peers/peer_book.rs

```diff
@@ -70,11 +71,11 @@ impl From<SerializedPeerBook> for PeerBook {
 #[derive(Debug, Default)]
 pub struct PeerBook {
     /// The map of the addresses currently being handshaken with.
-    connecting_peers: RwLock<HashSet<SocketAddr>>,
+    connecting_peers: MpmcMap<SocketAddr, Instant>,
     /// The map of connected peers to their metadata.
-    connected_peers: RwLock<HashMap<SocketAddr, PeerInfo>>,
+    connected_peers: MpmcMap<SocketAddr, PeerInfo>,
```

Comment:
I'm wondering if we shouldn't keep locks over these 3 collections, or at least change them at a later time; today I noticed that due to the async nature of connections we sometimes attempt to set a peer as `connected` more than once, which is fixed in https://github.com/AleoHQ/snarkOS/pull/779 via synchronous checks for the peer's status

Reply:
After running a node on this branch for a while I've noticed that the mismatch between the numbers of connected/disconnected peers has grown compared to the current setup, so I'd propose to postpone changing these 3 particular instances of `RwLock` to `MpmcMap` until later, preferably after https://github.com/AleoHQ/snarkOS/pull/779.

Reply:
The same happens to the calculation of `connecting` peers (and outside of metrics), which can break connection logic.